### PR TITLE
Systemtest python plugin dir

### PR DESCRIPTION
**Pull request by astoorangi**
The test checks if a python module logs events from the director correctly.

Added "python-modules" functionality

Every system test with a "python-module" folder will get gets the default python modules from bareos

### CODE REVIEW

File: systemtests/tests/python-dir-plugin-test/testrunner [hunk 1]
```diff
@@ -43,23 +43,15 @@ status client
...
 run_bareos
 check_for_zombie_jobs storage=File
 stop_bareos
 
-check_two_logs
-list="$(find ${BackupDirectory} -type f)"
-check_restore_only_files_diff $list
+if ! grep -Eq "^[bDirEventJobStart,bDirEventJobInit,bDirEventJobRun,bDirEventJobEnd]${1}" ${tmp}/test-plugin.log; then
+  estat=1
+fi
```
Comment (joergsteffens) [inline systemtests/tests/python-dir-plugin-test/testrunner]: Your grep statement looks wrong. How about the following one:

```suggestion
for i in bDirEventJobStart bDirEventJobInit bDirEventJobRun bDirEventJobEnd; do
  if ! grep "$i" ${tmp}/test-plugin.log; then
    set_error "Failed to find logged event $i"
  fi
done
```

Comment (joergsteffens) [reply systemtests/tests/python-dir-plugin-test/testrunner]: Beside the simple change to the "grep" line, I agree to this PR, but did ask @franku to also review this.
One other thing: please reformulate your 2. commit message summary to something like:

systemtests: added python-dir-plugin-test

File: systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  elif event == bDirEventType['bDirEventJobRun']:
+    toFile("bDirEventJobRun\n")
+
+  return bRCs['bRC_OK']
+
+def toFile(text):
+  doc = open(outputfile,"a")
+  doc.write(text)
+  doc.close()
```
Comment (arogge) [inline systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: while most things in Bareos are indented with 2 spaces, python has an official style guide (PEP-8) that has been applied to the rest of our Python code.
Consider reformatting using `black` or at least make sure your indentation uses 4 spaces.

File: systemtests/scripts/functions [hunk 3]
```diff
@@ -288,7 +288,6 @@ set_error()
 {
     estat=9
     echo "ERROR: $*" >> "$tmp/err.log"
-    echo "ERROR: $*"
```
[the remark this answers is not in the document]
Comment (astoorangi) [reply systemtests/scripts/functions]: end_test(), which is called at the end of each test, prints the "$tmp/err.log" file, so the error would be shown twice.

File: systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py (hunk 2 above)
Comment (astoorangi) [reply systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: I adopted the indentation from the templates in the core. Should I reformat them with black too?

https://github.com/bareos/bareos/blob/master/core/src/plugins/dird/bareos-dir.py.template

File: systemtests/CMakeLists.txt [hunk 4]
```diff
@@ -97,7 +97,6 @@ set(mon_dir_password "mon_dir_password")
 set(mon_fd_password "mon_fd_password")
 set(mon_sd_password "mon_sd_password")
 set(python_plugin_module_src_dir ${CMAKE_SOURCE_DIR}/core/src/plugins)
-set(plugindirtmp ${PROJECT_BINARY_DIR}/plugindirtmp)
```
[the remark this answers is not in the document]
Comment (astoorangi) [reply systemtests/CMakeLists.txt]: No, it is not required. I re-added the variable

File: systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py (hunk 2 above)
Comment (joergsteffens) [reply systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: You could do so, but please use a separate commit or even a separate PR.

File: systemtests/scripts/functions (hunk 3 above)
[the remark this answers is not in the document]
Comment (arogge) [reply systemtests/scripts/functions]: While we shouldn't print the message twice, we also should not delay its output till the test is finished.
For now I'd like to keep the old behaviour. Maybe we can discuss improvements to the systemtest logging in general later.

File: systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py (hunk 2 above)
Comment (arogge) [reply systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: A separate commit would be good for that, yes.

Comment (astoorangi) [reply systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: Should I reformat all python scripts in the repository or just the templates for the plugins?

Comment (astoorangi) [reply systemtests/tests/python-dir-plugin-test/python-modules/bareos-dir-test.py]: PR: #322

**Comment by joergsteffens** [comment]
I agree to merging this PR.
@arogge will you do the merging?

**Pull request merged**